test/src/index/transport/testBrowserstackTransport.js

```diff
@@ -228,25 +228,26 @@ xdescribe('BrowserstackTransport', function () {
     result.sessionId = '1234567';
     client.emit('nightwatch:session.create', result);
 
-    setTimeout(async function() {
-
-      nock('https://api.browserstack.com')
-        .put('/automate/sessions/1234567.json', {
-          status: 'failed',
-          reason: 'NightwatchAssertError: Timed out while waiting for element <#james> to be present for 5000 milliseconds. - expected "visible" but got: "not found" (5400ms)'
-        })
-        .reply(200, {});
-
-      const error = new Error('Timed out while waiting for element <#james> to be present for 5000 milliseconds. - expected "visible" but got: "not found" (5400ms)');
-      error.name = 'NightwatchAssertError';
-
-      result = await transport.testSuiteFinished(error);
-
-      assert.strictEqual(result, true);
-      assert.strictEqual(transport.sessionId, null);
-
-    }, 100);
+    return new Promise((resolve, reject) => {
+      setTimeout(async function() {
 
+        nock('https://api.browserstack.com')
+          .put('/automate/sessions/1234567.json', {
+            status: 'failed',
+            reason: 'NightwatchAssertError: Timed out while waiting for element <#james> to be present for 5000 milliseconds. - expected "visible" but got: "not found" (5400ms)'
+          })
+          .reply(200, {});
+  
+        const error = new Error('Timed out while waiting for element <#james> to be present for 5000 milliseconds. - expected "visible" but got: "not found" (5400ms)');
+        error.name = 'NightwatchAssertError';
+  
+        result = await transport.testSuiteFinished(error);
+  
+        assert.strictEqual(result, true);
+        assert.strictEqual(transport.sessionId, null);
```

Comment:
We need to handle the case for assertion failure here. Adding try/catch block and rejecting the promise from `catch` block will do that.

Reply:
Valid point.